relay-server/src/services/spooler/mod.rs

```diff
@@ -342,9 +335,9 @@ impl InMemory {
     }
 
     /// Dequeues the envelopes from the in-memory buffer and send them to provided `sender`.
-    fn dequeue(&mut self, keys: &Vec<QueueKey>, sender: mpsc::UnboundedSender<ManagedEnvelope>) {
+    fn dequeue(&mut self, keys: HashSet<QueueKey>, sender: mpsc::UnboundedSender<ManagedEnvelope>) {
```

Comment:
nit: I wonder if we should type this as an iterator and not care about the actual data structure?